nmoscommon/httpserver.py

```diff
@@ -17,74 +17,72 @@
 from geventwebsocket.handler import WebSocketHandler
 from gevent.pywsgi import WSGIServer
 import threading
-from flask import request
 import socket
 
 
 class HttpServer(threading.Thread):
-  daemon = True    # causes server to stop on main thread quitting
-  def __init__(self, api, port=0, host='0.0.0.0', api_args=None, api_kwargs=None, ssl=None):
-    self.api_args = api_args
-    self.api_kwargs = api_kwargs
-    if self.api_args is None:
-      self.api_args = []
-    if self.api_kwargs is None:
-      self.api_kwargs = {}
-    self.api_class = api
-    self.port = port
-    self.host = host
-    self.started = threading.Event()
-    self.failed = None
-    self.ssl = ssl
 
-    threading.Thread.__init__(self)
+    daemon = True    # causes server to stop on main thread quitting
 
-  def run(self):
-    self.api = self.api_class(*self.api_args, **self.api_kwargs)
+    def __init__(self, api, port=0, host='0.0.0.0', api_args=None, api_kwargs=None, ssl=None):
+        self.api_args = api_args
+        self.api_kwargs = api_kwargs
+        if self.api_args is None:
```

Comment:
Why not just do this in the default arg? i.e api_args=[] in the def statement?

Reply:
There's a Python gotcha here: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments